packages/flutter/lib/src/material/drawer.dart

```diff
@@ -198,12 +203,24 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
   }
 
   void _handleDragCancel() {
-    if (_controller.isDismissed || _controller.isAnimating)
-      return;
-    if (_controller.value < 0.5) {
-      close();
+    if (widget.type == DrawerType.START) {
+      if (_controller.isDismissed || _controller.isAnimating)
+        return;
+      if (_controller.value < 0.5) {
+        close();
+      } else {
+        open();
+      }      
     } else {
-      open();
+      if (_controller.isDismissed || _controller.isAnimating)
+        return;
+      if (_controller.value < 0.5) {
+        open();
+      } else if (_controller.value == 1.0) {
+        return;
+      } else {
+        close();
+      } 
```

Comment:
DrawerController already knows how to flip around based on the reading direction.  Rather than plumb this logic in this way, maybe it would make sense to flip the drawer controller around in the same way.

Specifically, I'd recommend keeping `_controller.value` logically "closed" and just change the visual presentation of the logical state.  That way `_controller.isDismissed` still corresponds to the drawer being closed, which makes more sense semantically.

Reply:
Fixed